src/connectionController.ts

```diff
@@ -143,6 +143,13 @@ export default class ConnectionController {
   }
 
   async loadSavedConnections(): Promise<void> {
+    try {
```

Comment:
You can check if we have something to migrate and only then attempt to requite keytar: https://github.com/mongodb-js/vscode/pull/572#discussion_r1286274579

Reply:
Moved to a check after connectionIdsThatDidNotMigrateEarlier.

Reply:
It is not the same :) I would rename the existing `connectionIdsThatDidNotMigrateEarlier` to `connectionIdsThatFailedMigrationEarlier`, because this is actually what happened: we tried to migrate, but it failed.

```typescript
connectionInfo.secretStorageLocation === SecretStorageLocation.Keytar
```

 You need a different check for keytar: check that we have never tried to migrate some connections.

```typescript
const hasConnectionsThatDidNotMigrateEarlier =
      !!globalAndWorkspaceConnections.some(([, connectionInfo]) => !connectionInfo.storageLocation);
```

Reply:
Pushed the change 😄